pkgs/tools/wayland/hyprpaper/default.nix

```diff
@@ -0,0 +1,58 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, pkg-config
+, gitMinimal
+, libjpeg
+, mesa
+, pango
+, wayland
+, wayland-protocols
+, wayland-scanner
+}:
+
+stdenv.mkDerivation rec {
+  pname = "hyprpaper";
+  version = "unstable-2022-07-04";
+
+  src = fetchFromGitHub {
+    owner = "hyprwm";
+    repo = pname;
+    rev = "e15912e9817d79bb988085c88e313fac5ab60940";
+    leaveDotGit = true;
```

Comment:
This is non-deterministic. Apparently. Sometimes. 

If there's a simple `substituteInPlace` you could add to hardcode the `version` to the Nix one that'd be nice.

Reply:
I'm only patching in the commit hash as I don't think there is a way to get the commit message without `.git`.